exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/registry/FunctionRegistryHolder.java

```diff
@@ -161,6 +163,38 @@ public void removeJar(String jarName) {
     }
   }
 
+  /**
+   * Retrieves all functions (holders) associated with all the jars
+   * This is read operation, so several users can perform this operation at the same time.
+   * @return list of all functions, mapped by their sources
+   */
+  public Map<String, List<FunctionHolder>> getAllJarsWithFunctionHolders() {
+    Map<String, List<FunctionHolder>> allFunctionHoldersByJar = new HashMap<>();
+
+    try (@SuppressWarnings("unused") Closeable lock = readLock.open()) {
+      for (String jarName : jars.keySet()) {
+        //Capture functionHolders here
+        List<FunctionHolder> drillFuncHolderList = new LinkedList<>();
```

Comment:
Why LinkedList?

Reply:
Since the variable is only used to add and remove the function holders, this seemed most efficient for memory and speed. Some sources (like `built-in`) will have 2K+ functions, while UDFs will have as little as 1 function.